### PR TITLE
refactor to check_gitlabuser() method

### PR DESCRIPTION
**Pull request by ychai23**

### CODE REVIEW

**Comment by ychai23** [comment]
@ktdreyer I created this to resolve [#26](url)

File: test_tar_changes.py [hunk 1]
```diff
@@ -24,7 +25,7 @@ def test_clear_old_changes_sources(tmpdir, monkeypatch):
     tarchanges.clear_old_changes_sources()
     sources = tmpdir.join('sources')
     expected = '3a393d427d5b16c33cf24da91244cc7a  ceph-12.2.8.tar.gz\n'
-    assert sources.read() == expected
+    assert sources.read() == expected, "clear_old_changes_sources() failed"
```
Comment (ktdreyer) [inline test_tar_changes.py:28]: Please take this out of this PR. Pytest will tell you what test is failing

File: test_tar_changes.py [hunk 2]
```diff
@@ -41,13 +42,15 @@ def distgitdir(tmpdir):
 
 
 def test_commit_distgit_amend(distgitdir):
-    suffix = 'GitLab-User: developer@example.com'
+    # Test if commit_distgit_amend(suffix) returns the correct formatted git commit msg
+    # suffix = tarchanges.check_gitlabuser()
+    suffix = "GitLab-User: developer@example.com"
```
Comment (ktdreyer) [inline test_tar_changes.py:47]: When the code is commented out like this on its own line, it's difficult to tell at a glance if it was a mistake.

What if we moved the comment onto the line it's describing, like this:

```suggestion
    suffix = "GitLab-User: developer@example.com"  # eg. from check_gitlabuser()
```

File: test_tar_changes.py [hunk 3]
```diff
@@ -41,13 +42,15 @@ def distgitdir(tmpdir):
...
     rng = 'HEAD~..HEAD'
     with distgitdir.as_cwd():
         tarchanges.commit_distgit_amend(suffix)
         message = git('log', '--format=%s%n%n%b', rng, log_cmd=False)
-    expected = """\
+    matched = message.startswith("""\
```
Comment (ktdreyer) [inline test_tar_changes.py:52]: It's unclear to me why we need to change to `.startswith()` here, instead of matching the whole string.

Can we go back to the simple string matching?

File: test_tar_changes.py [hunk 4]
```diff
@@ -4,6 +4,7 @@
 from rdopkg.utils.git import git
 import pytest
 import py.path
+import re
```
Comment (ktdreyer) [inline test_tar_changes.py:7]: We won't need this import any more.

File: test_tar_changes.py [hunk 5]
```diff
@@ -50,4 +52,4 @@ def test_commit_distgit_amend(distgitdir):
 second commit
 
 GitLab-User: developer@example.com"""
-    assert message == expected
+    assert message == expected
```
Comment (ktdreyer) [inline test_tar_changes.py:55]: Could you fix your editor to avoid removing the newline from this file?

**Comment by ktdreyer** [comment]
There are several distinct logical changes in this PR. Please rewrite your branch so this is easier to review and reason about. You can do this with `git rebase -i ...`. I'd like to see one commit per change, and one change per commit. Specifically:

1. One commit should refactor to your new `check_gitlabuser()` method.
2. One commit should add your comment to the top of `test_commit_distgit_amend()` method
3. Please take out the single-quote to double-quote `'` -> `"` change, since that's orthogonal to the rest of the PR.

File: rdopkg_tar/tar_changes.py [hunk 6]
```diff
@@ -123,6 +123,14 @@ def commit_distgit_amend(suffix):
     cmd = ['commit', '-a', '-F', '-', '--amend']
     git(*cmd, input=message, print_output=True)
 
+def check_gitlabuser():
+    # Check for the original commiter username in jenkins env variables.
```
Comment (ktdreyer) [inline rdopkg_tar/tar_changes.py:127]: I thought your earlier change that used `"""` comments for docstrings was good. Why did you move back to using `#` comments? You can go back to `"""` if you want..

File: rdopkg_tar/tar_changes.py [hunk 7]
```diff
@@ -123,6 +123,14 @@ def commit_distgit_amend(suffix):
...
+    # If it exists, then amend a suffix to the commit.
+    userName = os.environ.get('gitlabUserName')
+    gitlabuserName = os.environ.get('gitlabUserUsername')
+
+    if userName or gitlabuserName:
+        commit_distgit_amend(suffix='GitLab-User: ' + gitlabuserName + ' (' + userName + ')')
```
Comment (ktdreyer) [inline rdopkg_tar/tar_changes.py:133]: One of these two commits changes this line from double-quotes `"` to `'` single-quotes. Please amend that commit to avoid this because it does not relate to the commit. "One (logical) change per commit, one commit per change." is a lot simpler and easier to review.

File: test_tar_changes.py [hunk 8]
```diff
@@ -41,6 +41,7 @@ def distgitdir(tmpdir):
 
 
 def test_commit_distgit_amend(distgitdir):
+    # Test commit_distgit_amend()
```
Comment (ktdreyer) [inline test_tar_changes.py:44]: A comment like `Test commit_distgit_amend()` is not adding any value - the method name is literally `test_commit_distgit_amend()`.

If there's something a casual reader would need to know beyond the method name, then let's add it here. Otherwise, we don't need to add anything.

**Comment by ktdreyer** [comment]
I'm not sure this PR has anything to do with #26 any more.

It's good to refactor to a separate `check_gitlabuser()` method, though. That code badly needs unit tests, and it will be much easier to unit test once you're refactored it.

**Comment by ktdreyer** [comment]
Please read https://fagnerbrack.com/one-pull-request-one-concern-e84a27dfe9f1

The `adjust comments` commit does not only adjust comments. It also changes the logical behavior of the code. Please don't do that in one commit, and please do not add multiple distinct changes into this PR.

Let's slow down. Let's decide what this the purpose of this PR is. There are many different changes coming and going here.

I think this PR should only do one thing: refactor to `check_gitlabuser()`. It should preserve rdopkg-tar's existing behavior, (including preserving the bug in #33 ). Once we've merged this PR, then we can make other changes.

**Comment by ychai23** [comment]
I rebased the commits and now it only resolves the refactor method.

File: rdopkg_tar/tar_changes.py [hunk 9]
```diff
@@ -123,6 +123,16 @@ def commit_distgit_amend(suffix):
...
+def check_gitlabuser():
+    """
+    Check for the original commiter username in jenkins env variables.
+    If it exists, then amend a suffix to the commit.
+    """
+    userName = os.environ.get('gitlabUserName')
+    gitlabuserName = os.environ.get('gitlabUserUsername')
+
+    if (userName and gitlabuserName):
+        commit_distgit_amend(suffix='GitLab-User: ' + gitlabuserName + ' (' + userName + ')')
```
Comment (ktdreyer) [inline rdopkg_tar/tar_changes.py:135]: Please read https://www.codewithjason.com/dont-mix-refactorings-behavior-changes/

When refactoring, please preserve the old behavior *exactly*:
```suggestion
        commit_distgit_amend(suffix="GitLab-User: " + gitlabuserName + " " + userName)
```

Comment (ychai23) [reply rdopkg_tar/tar_changes.py:135]: So would you like to change this format later in another MR? Or we can keep it this way.

Comment (ychai23) [reply rdopkg_tar/tar_changes.py:135]: Would you like to make the change in another MR or we can preserve the old behavior?

Comment (ktdreyer) [reply rdopkg_tar/tar_changes.py]: Let's preserve the old behavior for now.

File: rdopkg_tar/tar_changes.py [hunk 10]
```diff
@@ -123,6 +123,16 @@ def commit_distgit_amend(suffix):
...
 
+def check_gitlabuser():
+    """
+    Check for the original commiter username in jenkins env variables.
+    If it exists, then amend a suffix to the commit.
+    """
+    userName = os.environ.get('gitlabUserName')
+    gitlabuserName = os.environ.get('gitlabUserUsername')
+
+    if (userName and gitlabuserName):
```
Comment (ktdreyer) [inline rdopkg_tar/tar_changes.py:134]: Please keep this behavior the same too:
```suggestion
    if userName or gitlabuserName:
```

File: rdopkg_tar/tar_changes.py [hunk 11]
```diff
@@ -123,6 +123,16 @@ def commit_distgit_amend(suffix):
...
 
+def check_gitlabuser():
+    """
+    Check for the original commiter username in jenkins env variables.
+    If it exists, then amend a suffix to the commit.
+    """
+    userName = os.environ.get('gitlabUserName')
+    gitlabuserName = os.environ.get('gitlabUserUsername')
+
+    if (userName or gitlabuserName):
```
Comment (ktdreyer) [inline rdopkg_tar/tar_changes.py:134]:
```suggestion
    if userName or gitlabuserName:
```

File: rdopkg_tar/tar_changes.py [hunk 12]
```diff
@@ -201,12 +211,8 @@ def main():
 
     # Check for the original commiter username in jenkins env variables.
     # If it exists, then amend a suffix to the commit.
```
Comment (ktdreyer) [inline rdopkg_tar/tar_changes.py:213]: Let's remove this comment, since it's now duplicated (copy-pasted) in the method above:
```suggestion
```

File: rdopkg_tar/tar_changes.py [hunk 13]
```diff
@@ -199,14 +209,8 @@ def main():
...
-    # Check for the original commiter username in jenkins env variables.
-    # If it exists, then amend a suffix to the commit.
-    userName = os.environ.get('gitlabUserName')
-    gitlabuserName = os.environ.get('gitlabUserUsername')
-
-    if userName or gitlabuserName:
-        commit_distgit_amend(suffix="GitLab-User: " + gitlabuserName + " " + userName)
-
+    check_gitlabuser()
+
```
Comment (ktdreyer) [inline rdopkg_tar/tar_changes.py:213]: Please remove the trailing whitespace here:

```suggestion

```

**Review by ktdreyer — CHANGES_REQUESTED** [review]
Please amend your Git commit log with `git commit --amend` and remove the "changes", "keep" ,"keep", "remove" bits of text.
